cpp/density/GaussianDensity.cc

```diff
@@ -81,44 +81,51 @@ void GaussianDensity::compute(const freud::locality::NeighborQuery* nq)
                 bin_z = 0;
             }
 
+            // Reject bins that are outside the box in aperiodic directions
             // Only evaluate over bins that are within the cutoff
             for (int k = bin_z - bin_cut_z; k <= bin_z + bin_cut_z; k++)
             {
+                if (!periodic.z && (k < 0 || k >= int(m_width.z)))
+                {
+                    continue;
+                }
                 const float dz = float((grid_size_z * k + grid_size_z / 2.0f) - point.z - lz / 2.0f);
 
                 for (int j = bin_y - bin_cut_y; j <= bin_y + bin_cut_y; j++)
                 {
+                    if (!periodic.y && (j < 0 || j >= int(m_width.y)))
+                    {
+                        continue;
+                    }
                     const float dy = float((grid_size_y * j + grid_size_y / 2.0f) - point.y - ly / 2.0f);
 
                     for (int i = bin_x - bin_cut_x; i <= bin_x + bin_cut_x; i++)
                     {
-                        // Calculate the distance from the particle to the grid cell
+                        if (!periodic.x && (i < 0 || i >= int(m_width.x)))
+                        {
+                            continue;
+                        }
                         const float dx = float((grid_size_x * i + grid_size_x / 2.0f) - point.x - lx / 2.0f);
-                        vec3<float> delta = m_box.wrap(vec3<float>(dx, dy, dz));
+
+                        // Calculate the distance from the particle to the grid cell
+                        const vec3<float> delta = m_box.wrap(vec3<float>(dx, dy, dz));
 
                         const float r_sq = dot(delta, delta);
-                        const float r_sqrt = std::sqrt(r_sq);
 
                         // Check to see if this distance is within the specified r_max
-                        if (r_sqrt < m_r_max)
+                        if (r_sq < r_max_sq)
                         {
-                            // Evaluate the gaussian ...
-                            const float x_gaussian
-                                = A * std::exp((-1.0f) * (delta.x * delta.x) / (2.0f * sigmasq));
-                            const float y_gaussian
-                                = A * std::exp((-1.0f) * (delta.y * delta.y) / (2.0f * sigmasq));
-                            const float z_gaussian
-                                = A * std::exp((-1.0f) * (delta.z * delta.z) / (2.0f * sigmasq));
+                            // Evaluate the gaussian
+                            const float gaussian = A * std::exp(-r_sq / (2 * sigmasq));
```

Comment:
There is a bug in the Gaussian normalization in the old code for 2D systems. I am adding a separate changelog line to address that.